beater/beater.go

```diff
@@ -280,7 +285,7 @@ type sharedServerRunnerParams struct {
 }
 
 func newServerRunner(ctx context.Context, args serverRunnerParams) (*serverRunner, error) {
-	cfg, err := config.NewConfig(args.RawConfig, elasticsearchOutputConfig(args.Beat))
+	cfg, err := config.NewConfig(args.RawConfig, args.KibanaConfig, elasticsearchOutputConfig(args.Beat))
 	if err != nil {
 		return nil, err
 	}
```

Comment:
```suggestion
	cfg, err := config.NewConfig(args.RawConfig, elasticsearchOutputConfig(args.Beat))
	if err != nil {
		return nil, err
	}
	if args.KibanaConfig != nil {
		args.Kibana = *args.KibanaConfig
	}
```

Seeing as this is the only place we pass in the arg, can we just replace the Kibana config afterwards and avoid forcing all other callers to pass in `nil`? Alternatively we could not unpack `Fleet.Kibana` in the integration config, and merge it into the config in `serverCreator.Create`.